### PR TITLE
Update json_ref

### PR DESCRIPTION
**Pull request by bamurtaugh**
Update based on https://github.com/devcontainers/spec/pull/122, https://github.com/devcontainers/spec/pull/124, and a variety of other changes in appears weren't aligned between containers.dev and devcontainers/spec

@Chuxel I just noticed https://github.com/devcontainers/devcontainers.github.io/pull/51 so I was going to delete this branch, but as I found and addressed some other ordering and wording inconsistencies in this branch, maybe we merge this PR instead if it looks good to you? 

I also noticed a section missing from devcontainers/spec as I was diffing these two files, so opened this PR to update: https://github.com/devcontainers/spec/pull/126.

### CODE REVIEW

**Comment by Chuxel** [comment]
This one looks good!  I can close my PR.

**Review by Chuxel — APPROVED** [review]

File: _implementors/json_reference.md [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
-title:  "devcontainer.json reference"
-shortTitle: "devcontainer.json reference"
+title:  "Dev Container metadata reference"
+shortTitle: "Dev Container metadata reference"
 author: Microsoft
 index: 4
 ---
 
-A `devcontainer.json` file in your project tells [tools and services that support the dev container spec](../../supporting) how to access (or create) a **development container** with a well-defined tool and runtime stack. It follows the JSON with Comments (jsonc) format.
+The `devcontainer.json` file contains any needed metadata and settings required to configurate a **development container** for a given well-defined tool and runtime stack. It can be used by [tools and services that support the dev container spec](../../supporting) to create a **development environment** that contains one or more **development containers**.
```
Comment (samruddhikhandale) [inline _implementors/json_reference.md:9]: This ref returns a 404 when opened in a preview 🤔 

https://github.com/devcontainers/devcontainers.github.io/blob/54355ca4da67096df87fb697bef80da5bcf58f8c/_implementors/json_reference.md

**Review by samruddhikhandale — COMMENTED** [review]
left one comment! 
Besides that looks great.

File: _implementors/json_reference.md (hunk 1 above)
Comment (bamurtaugh) [reply _implementors/json_reference.md:9]: The supporting link? I know it's odd how the site formats links in preview. I used the same link that is [currently in the top paragraph](https://containers.dev/implementors/json_reference/), and that link works, so I think this should be ok. Let me know if I might be missing anything.

Comment (samruddhikhandale) [reply _implementors/json_reference.md:9]: umm. good to know.

**Review by samruddhikhandale — APPROVED** [review]
🚀

**Comment by bamurtaugh** [comment]
Thank you both! 🚀

**Pull request merged**